### PR TITLE
use different secrets for pypi prod and test

### PR DESCRIPTION
**Pull request by beniwohli**

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -188,7 +188,13 @@ def generateStep(version){
 }
 
 def releasePackages(){
-  withSecretVault(secret: 'secret/apm-team/ci/apm-agent-python-twine',
+  def vault_secret = ''
```
Comment (mdelapenya) [inline .ci/Jenkinsfile]: Just a matter of preference... why not?:
```groovy
def vault_secret = 'secret/apm-team/ci/apm-agent-python-pypi-test'
if (env.REPO_URL == 'https://upload.pypi.org/legacy/'){
    vault_secret = 'secret/apm-team/ci/apm-agent-python-pypi-prod'
}
```

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -188,7 +188,13 @@ def generateStep(version){
 }
 
 def releasePackages(){
-  withSecretVault(secret: 'secret/apm-team/ci/apm-agent-python-twine',
```
Comment (mdelapenya) [inline .ci/Jenkinsfile]: What happen with this vault key, `secret/apm-team/ci/apm-agent-python-twine`? Must be updated/revoked?

Comment (beniwohli) [reply .ci/Jenkinsfile]: I'll delete that user on PyPI once we verified that the token auth works as intended. At that point, we can also delete it in the vault (or delete it now if that's easier, with the minimal risk that we'll have to re-add it if this doesn't work out)

File: .ci/Jenkinsfile (hunk 1 above)
Comment (beniwohli) [reply .ci/Jenkinsfile]: done in bfc1e0d :)

File: .ci/Jenkinsfile [hunk 3]
```diff
@@ -188,7 +188,11 @@ def generateStep(version){
...
+  def vault_secret = 'secret/apm-team/ci/apm-agent-python-pypi-test'
```
Comment (v1v) [inline .ci/Jenkinsfile]: More groovysh:
```suggestion
  def vault_secret = (env.REPO_URL == 'https://upload.pypi.org/legacy/') ? 'secret/apm-team/ci/apm-agent-python-pypi-prod' : 'secret/apm-team/ci/apm-agent-python-pypi-test'
```

The above suggestion replaces from 191...194

File: Jenkinsfile [hunk 4]
```diff
@@ -302,7 +302,11 @@ def runScript(Map params = [:]){
 }
 
 def releasePackages(){
-  withSecretVault(secret: 'secret/apm-team/ci/apm-agent-python-twine',
+  def vault_secret = 'secret/apm-team/ci/apm-agent-python-pypi-test'
+  if (env.REPO_URL == 'https://upload.pypi.org/legacy/'){
```
Comment (v1v) [inline Jenkinsfile]: ditto 

```suggestion
  vault_secret = (env.REPO_URL == 'https://upload.pypi.org/legacy/') ? 'secret/apm-team/ci/apm-agent-python-pypi-prod' : 'secret/apm-team/ci/apm-agent-python-pypi-test'
```

File: .ci/Jenkinsfile [hunk 5]
```diff
@@ -188,7 +188,11 @@ def generateStep(version){
 }
 
 def releasePackages(){
-  withSecretVault(secret: 'secret/apm-team/ci/apm-agent-python-twine',
+  def vault_secret = (env.REPO_URL == 'https://upload.pypi.org/legacy/') ? 'secret/apm-team/ci/apm-agent-python-pypi-prod' : 'secret/apm-team/ci/apm-agent-python-pypi-test'
+  if (env.REPO_URL == 'https://upload.pypi.org/legacy/'){
```
Comment (mdelapenya) [inline .ci/Jenkinsfile]: After @v1v's comments, lines L192:194 are not needed, as L191 uses the ternary operator to evaluate and assign the variable

File: Jenkinsfile [hunk 6]
```diff
@@ -302,7 +302,11 @@ def runScript(Map params = [:]){
 }
 
 def releasePackages(){
-  withSecretVault(secret: 'secret/apm-team/ci/apm-agent-python-twine',
+  def vault_secret = 'secret/apm-team/ci/apm-agent-python-pypi-test'
```
Comment (mdelapenya) [inline Jenkinsfile]: This block (L305:309) is wrong. Please replace it with:

```groovy
def vault_secret = (env.REPO_URL == 'https://upload.pypi.org/legacy/') ? 'secret/apm-team/ci/apm-agent-python-pypi-prod' : 'secret/apm-team/ci/apm-agent-python-pypi-test'

```

File: .ci/Jenkinsfile (hunk 5 above)
Comment (beniwohli) [reply .ci/Jenkinsfile]: yeah, this whole "suggested changes" thing in github comments didn't really do what I think it would do. Thanks for catching it!

**Pull request merged**